### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by AndriyUrban**
Issue #4487

### CODE REVIEW

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 1]
```diff
@@ -626,4 +629,79 @@ void updateShoppingItem() {
...
+        List<HabitAssignDto> expectedList = List.of(hd);
+        habitAssign.setUserShoppingListItems(Collections.singletonList(getUserShoppingListItem()));
+        when(habitAssignRepo.findAllByUserId(1L)).thenReturn(Collections.singletonList(habitAssign));
+        when(modelMapper.map(habitAssign, HabitAssignDto.class)).thenReturn(hd);
+        when(modelMapper.map(habitAssign, HabitAssignDto.class)).thenReturn(hd);
+        when(modelMapper.map(HabitTranslation.builder().id(1L).name("").description("").habitItem("")
+            .language(Language.builder().code("en").id(1L)
+                .build())
+            .build(), HabitDto.class)).thenReturn(habitDto);
+        System.out.println();
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Please remove it

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 2]
```diff
@@ -626,4 +629,79 @@ void updateShoppingItem() {
...
+        hd.setUserShoppingListItems(Collections.emptyList());
+        hd.getHabit().setShoppingListItems(Collections.emptyList());
+        hd.setCreateDateTime(time);
+        List<HabitAssignDto> expectedList = List.of(hd);
+        habitAssign.setUserShoppingListItems(Collections.singletonList(getUserShoppingListItem()));
+        when(habitAssignRepo.findAllByUserId(1L)).thenReturn(Collections.singletonList(habitAssign));
+        when(modelMapper.map(habitAssign, HabitAssignDto.class)).thenReturn(hd);
+        when(modelMapper.map(habitAssign, HabitAssignDto.class)).thenReturn(hd);
+        when(modelMapper.map(HabitTranslation.builder().id(1L).name("").description("").habitItem("")
+            .language(Language.builder().code("en").id(1L)
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java:652]: Please move the creation of object from constructor

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 3]
```diff
@@ -626,4 +629,79 @@ void updateShoppingItem() {
...
+            .build(), HabitDto.class)).thenReturn(habitDto);
+        System.out.println();
+        assertEquals(expectedList, habitAssignService.getAllCustomHabitAssignsByUserId(1L, language));
+    }
+
+    @Test
+    void changeStatuses() {
+        try {
+            Method method =
+                habitAssignService.getClass().getDeclaredMethod("changeStatuses", String.class, Long.class, List.class);
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Why do you use the here Reflection API?

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 4]
```diff
@@ -626,4 +629,79 @@ void updateShoppingItem() {
...
 
     }
+
+    @Test
+    void saveUserShoppingListItems() {
+        List<ShoppingListItem> shoppingList = new ArrayList<>();
+        try {
+            Method method = habitAssignService
+                .getClass().getDeclaredMethod("saveUserShoppingListItems", List.class, HabitAssign.class);
+            method.setAccessible(true);
```
Comment (VadymMakitra) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Why do you use Reflection API? It's not recommended to do in the tests

File: service/src/main/java/greencity/service/HabitAssignServiceImpl.java [hunk 5]
```diff
@@ -278,7 +278,7 @@ private HabitAssign buildHabitAssign(Habit habit, User user) {
      * @param language    code of {@link Language}.
      * @return {@link HabitAssign} instance.
      */
-    private HabitAssignDto buildHabitAssignDto(HabitAssign habitAssign, String language) {
+    public HabitAssignDto buildHabitAssignDto(HabitAssign habitAssign, String language) {
```
Comment (VadymMakitra) [inline service/src/main/java/greencity/service/HabitAssignServiceImpl.java]: Why do you change the modification of access here?

**Review by VadymMakitra — CHANGES_REQUESTED** [review]
Please resolve comments

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java (hunk 4 above)
Comment (AndriyUrban) [reply service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: I haven't seen any PowerMock dependency in pom.xml, so I decided to use Reflection here.
In which way, I should fix it?

Comment (danylohlynsky) [reply service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Just add PowerMock dependency

Comment (RomaMocherniuk) [reply service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: It is bad practice to use reflection for testing private methods. Private methods should be tested using public methods.
Method `saveUserShoppingListItems` can be tested using `assignCustomHabitForUser` and `assignDefaultHabitForUser` methods where it is  used

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 6]
```diff
@@ -13,6 +13,8 @@
 import greencity.exception.exceptions.*;
 import greencity.repository.*;
 
+import java.lang.reflect.InvocationTargetException;
```
Comment (innayashna) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Please remove unused imports from reflection API

File: service/src/test/java/greencity/service/HabitAssignServiceImplTest.java [hunk 7]
```diff
@@ -626,4 +628,77 @@ void updateShoppingItem() {
...
+            .language(Language.builder().code("en").id(1L)
+                .build())
+            .build(), HabitDto.class)).thenReturn(habitDto);
+        assertEquals(expectedList, habitAssignService.getAllCustomHabitAssignsByUserId(1L, language));
+    }
+
+    @Test
+    void changeStatuses() {
+        findHabitByUserIdAndHabitId();
+//        when(habitAssignRepo.findByHabitIdAndUserId(1L, 1L)).thenReturn(Optional.of(habitAssign));
```
Comment (innayashna) [inline service/src/test/java/greencity/service/HabitAssignServiceImplTest.java]: Please remove the commented code

**Review by innayashna — CHANGES_REQUESTED** [review]

**Review by innayashna — APPROVED** [review]

**Review by OlegVat — APPROVED** [review]

**Review by VadymMakitra — APPROVED** [review]

**Pull request closed without merging**